connect/basic-auth-extension/src/main/java/org/apache/kafka/connect/rest/basic/auth/extension/PropertyFileLoginModule.java

```diff
@@ -62,35 +62,63 @@ public void initialize(Subject subject, CallbackHandler callbackHandler, Map<Str
         if (fileName == null || fileName.trim().isEmpty()) {
             throw new ConfigException("Property Credentials file must be specified");
         }
+
         if (!credentialPropertiesMap.containsKey(fileName)) {
+            log.trace("Opening credential properties file '{}'", fileName);
             Properties credentialProperties = new Properties();
             try {
                 try (InputStream inputStream = Files.newInputStream(Paths.get(fileName))) {
+                    log.trace("Parsing credential properties file '{}'", fileName);
                     credentialProperties.load(inputStream);
                 }
                 credentialPropertiesMap.putIfAbsent(fileName, credentialProperties);
+                if (credentialProperties.isEmpty())
+                    log.warn("Credential properties file '{}' is empty; all requests will be permitted",
+                        fileName);
             } catch (IOException e) {
                 log.error("Error loading credentials file ", e);
                 throw new ConfigException("Error loading Property Credentials file");
             }
+        } else {
+            log.trace(
+                "Credential properties file '{}' has already been opened and parsed; will read from cached, in-memory store",
+                fileName);
         }
     }
 
     @Override
     public boolean login() throws LoginException {
         Callback[] callbacks = configureCallbacks();
         try {
+            log.trace("Authenticating user; invoking JAAS login callbacks");
             callbackHandler.handle(callbacks);
         } catch (Exception e) {
+            log.trace("Authentication failed while invoking JAAS login callbacks");
             throw new LoginException(e.getMessage());
         }
 
         String username = ((NameCallback) callbacks[0]).getName();
         char[] passwordChars = ((PasswordCallback) callbacks[1]).getPassword();
         String password = passwordChars != null ? new String(passwordChars) : null;
         Properties credentialProperties = credentialPropertiesMap.get(fileName);
-        authenticated = credentialProperties.isEmpty() ||
-                        (password != null && password.equals(credentialProperties.get(username)));
+
+        if (credentialProperties.isEmpty()) {
+            log.trace("Not validating credentials for user '{}' as credential properties file '{}' is empty",
```

Comment:
This log is likely a duplicate of the one that we have in the `initialize`. Should we log this for every login?

Reply:
I think it's alright if it's at `TRACE` level. My thinking was that since some Connect clusters are left running for a long time, the original log line in `initialize` may be lost by the time someone wants to investigate the behavior of the extension.

Reply:
Typically, most users push logs to something like elastic/splunk and should be able to lookup the logs from the archive. I'm not too concerned about this since properties file based credentials store shouldn't typically be used in a production environment. So,I'm fine leaving the log statement in here. But let's see what the committers have to say fro this :).

Reply:
I think it's a pretty big assumption that users will have infinite (or at least very forgiving) log retention, though, and banking on that leaves ones who have more conservative policies out on their own.

Additionally, the log message in `initialize` would be sufficient for anyone who knew to look for it, but for anyone else, a log message that happens whenever a REST request that gets authenticated would be easier to spot.